apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -175,6 +181,55 @@ func validateDNSName(hostaddress string) error {
 	return nil
 }
 
+func validateAnnotations(host *BareMetalHost) []error {
+	var errs []error
+	var err error
+
+	for annotation, value := range host.Annotations {
+
+		switch {
+		case annotation == StatusAnnotation:
+			err = validStatusAnnotation(value)
+		case strings.HasPrefix(annotation, RebootAnnotationPrefix+"/") || annotation == RebootAnnotationPrefix:
+			err = validRebootAnnotation(value)
+		case annotation == InspectAnnotation:
+			err = validInspectAnnotation(value)
+		case annotation == HardwareDetailsAnnotation:
+			inspect := host.Annotations[InspectAnnotation]
+			err = validHwdDetailsAnnotation(value, inspect)
+		default:
+			err = nil
+		}
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func validStatusAnnotation(statusannotation string) error {
```

Comment:
nit: statusAnnotation

Reply:
made changes where the came case was not followed